core/registry/legacy.go

```diff
@@ -0,0 +1,14 @@
+package registry
+
+import gogoproto "github.com/cosmos/gogoproto/proto"
+
+type LegacyRegistry interface {
+	RegisterInterface(protoName string, iface interface{}, impls ...gogoproto.Message)
```

Comment:
```suggestion
	RegisterInterface(protoName string, iface interface{}, impls ... protoiface.MessageV1)
```

Reply:
This will take a bigger refactor though right? The concrete implementation we're passing around expects `gogoproto.Message` 
https://github.com/cosmos/cosmos-sdk/blob/aa9ff3d8a77fe2e4ab5463f135bf4a2693372a00/codec/types/interface_registry.go#L191-L196